static/js/buddy_list.js

```diff
@@ -115,17 +155,63 @@ var buddy_list = (function () {
         return self.keys.length;
     };
 
+    self.show_key = function (opts) {
+        var key = opts.key;
+
+        var pos = self.find_position({
+            key: key,
+        });
+
+        // Try to render a bit optimistically here.
+        var cushion_size = 3;
+        var chunk_size = pos + cushion_size - self.render_count;
```

Comment:
I should double check this is > 0.  (In theory it should be due to how we're called, but I want to be defensive.)